zkevm-circuits/src/evm_circuit/execution/end_tx.rs

```diff
@@ -70,6 +71,16 @@ impl<F: Field> ExecutionGadget<F> for EndTxGadget<F> {
         cb.tx_refund_read(tx_id.expr(), refund.expr());
         let effective_refund = MinMaxGadget::construct(cb, max_refund.quotient(), refund.expr());
 
+        // gas_used & refund == 0 if tx is invalid
+        cb.require_zero(
+            "is_tx_invalid * gas_used == 0",
+            is_tx_invalid.expr() * gas_used.expr(),
+        );
+        cb.require_zero(
+            "is_tx_invalid * effective_refund == 0",
+            is_tx_invalid.expr() * effective_refund.min().expr(),
+        );
```

Comment:
Use `cb.condition` so it's more clear this is a conditional constraint on `is_tx_invalid`, like you did below for the log id.